### DISCUSSION TITLE
RLS by fields

### DISCUSSION THREAD

**Discussion by flo-butterfly** [supabase/supabase#18621] [Questions] [answered]
Hi all,

I wonder how we can define  the RLS by fields.
Let's say you have a table profile with the following fields
- id
- firstname
- lastname
- email
- username
- picture

using JWT if:
- the user (owner) of the record does a select, all the record fields should be returned + allow updates
- an authenticated user does a select, return only username and picture + no update
- an unauthenticated user does a select, return only username + no update

Not sure if makes sense. Let me know if more information is needed

Thanks for your help

**Comment by GaryAustin1** [comment, marked as answer]
No.  RLS only works on rows.  There are Postgres column grants but those only work on authenticated or anon and would not work for your first case.

Consider a view or rpc call returning setof records or a table.  Or split out you columns to separate tables and use joins with different RLS on the tables.

**Reply by flo-butterfly** [reply]
Thanks for your answer @GaryAustin1 , regarding views, is there a place on the dash where to create them and review them?

**Reply by GaryAustin1** [reply]
You can see them in the table UI.  You have to create them in the SQL editor.

**Reply by flo-butterfly** [reply]
OK. Sorry for my lack of knowledge on the platform. The problem i see is that views are using the RLS of the table which means that in my previously mentioned schema, i would still have to let the table with read capabilities to anyone so it doesn't really make it any safer.

**Reply by GaryAustin1** [reply]
The view would check for user type and manipulate what columns are shown OR there would be separate views with only specific columns and those would also check in the where statements for the type of users or auth.uid().

**Reply by flo-butterfly** [reply]
thanks for your help @GaryAustin1 !

**Reply by GaryAustin1** [reply]
Might be useful.  I'm sure there is more on the web.  https://github.com/orgs/supabase/discussions/656
